### PR TITLE
PR2022-37/Layout-of-registration

### PR DESCRIPTION
**Pull request by Neoclassic-alt**
PR2022-37 add layout of registration user page

### CODE REVIEW

File: src/types/register/register.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export type RegisterUserData = {
```
Comment (VKhayretdinov) [inline src/types/register/register.ts:1]: If you open API spec you'll see that the same entity has the name User. My suggestion is to use names for UI model from API. It'll help us to work with entities easier

File: src/modules/Register/components/RegisterForm.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  const { control } = useForm({
+    defaultValues: DEFAULT_FORM_VALUES,
+  });
+
+  const history = useHistory();
+
+  /* const schema = yup.object({
+    name: yup.string().required('name is required'),
+    login: yup.string().email().required('e-mail must be a valid email'),
+    password: yup.string().min(8, 'password must be at least 8 characters long')}) */
```
Comment (VKhayretdinov) [inline src/modules/Register/components/RegisterForm.tsx:22]: Don't keep commented code. You can stash it if it's needed

**Review by VKhayretdinov — CHANGES_REQUESTED** [review]
I left just base thoughts about general stuff. And I don't see rendering of this component. You should leave additional information about your PR, at least.

File: src/types/register/register.ts (hunk 1 above)
Comment (NKonstantin007) [reply src/types/register/register.ts:1]: Vlad, the name of model in spec is SignUpCred. And it's really credentials not an user. I think SignUpCredentials or for instance RegistrationUserData is suitable names but not a User.
image

**Comment by Neoclassic-alt** [comment]
Changes are applied. Also added field "Change password" to the form.

File: src/types/registration/registration.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export type RegisterUserData = {
```
Comment (VKhayretdinov) [inline src/types/registration/registration.ts:1]: Please rename the types according to the previous discussion

File: src/types/registration/registration.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  name: String;
+  login: String;
+  password: String;
+};
+
+export type RegisterUserFormData = RegisterUserData & {
```
Comment (VKhayretdinov) [inline src/types/registration/registration.ts:7]: Are you sure you will use this type outside of the registration form? Move it to the module level if you don't

**Review by VKhayretdinov — CHANGES_REQUESTED** [review]

File: src/types/registration/registration.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export type RegisterUserData = {
+  name: String;
+  login: String;
+  password: String;
```
Comment (VKhayretdinov) [inline src/types/registration/registration.ts:4]:
```suggestion
  name: string;
  login: string;
  password: string;
```

**Review by VKhayretdinov — APPROVED** [review]

**Pull request merged**